pkg/fleet/internal/msi/msilog.go

```diff
@@ -0,0 +1,137 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+//go:build windows
+
+// Package msi contains helper functions to work with msi packages
+package msi
+
+import (
+	"bytes"
+	"fmt"
+	"golang.org/x/text/encoding/unicode"
+	"io"
+	"io/fs"
+	"regexp"
+)
+
+// FindAllIndexWithContext is similar to FindAllIndex but expands the matched range for a number of lines
+// before and after the match (called contextBefore and contextAfter).
+func FindAllIndexWithContext(r *regexp.Regexp, input []byte, contextBefore, contextAfter int) [][]int {
+	var extractedRanges [][]int
+	results := r.FindAllIndex(input, -1)
+	for _, result := range results {
+		lineCounter := 0
+		charCounter := result[0]
+		for ; charCounter >= 0; charCounter-- {
+			if input[charCounter] == '\n' {
+				lineCounter++
+			}
+			if lineCounter > contextBefore {
+				break
+			}
+		}
+		lineStart := charCounter + 1
+
+		lineCounter = 0
+		charCounter = result[1]
+		for ; charCounter < len(input); charCounter++ {
+			if input[charCounter] == '\n' {
+				lineCounter++
+			}
+			if lineCounter > contextAfter {
+				break
+			}
+		}
+		lineEnd := charCounter
+
+		extractedRanges = append(extractedRanges, []int{lineStart, lineEnd})
+	}
+
+	return extractedRanges
+}
+
+func insert(existingRanges, newRanges [][]int) [][]int {
+	for _, newRange := range newRanges {
+		matched := false
+		for _, existingRange := range existingRanges {
+			bnr := newRange[0]
+			enr := newRange[1]
+			ber := existingRange[0]
+			eer := existingRange[1]
+			// [ber, eer]
+			//   [bnr,  enr]
+			if bnr <= eer && eer < enr {
+				existingRange[1] = newRange[1]
+				matched = true
+			}
+			//   [ber, eer]
+			// [bnr,  enr]
+			if bnr <= ber && ber < enr {
+				existingRange[0] = newRange[0]
+				matched = true
+			}
+			//   [ber, eer]
+			// [bnr,      enr]
+			if bnr <= ber && enr >= eer {
+				existingRange[0] = newRange[0]
+				existingRange[1] = newRange[1]
+				matched = true
+			}
+			// [ber,    eer]
+			//   [bnr, enr]
+			if ber <= bnr && enr <= eer {
+				matched = true
+			}
+		}
+		if !matched {
+			existingRanges = append(existingRanges, newRange)
```

Comment:
you're modifying the range that you're iterating over. The `range` will not include the new elements, is that what you want?